app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -161,6 +163,26 @@
     private boolean cancelled;
     private ContainerFragment<EntityListAdapter> containerFragment;
 
+    // Function handler for handling XPath evaluation of the function here().
+    // Although only one instance is created, which is used by NodeEntityFactory,
+    // every instance of EntitySelectActivity registers itself (one at a time)
+    // to listen to the handler and refresh whenever a new location is obtained.
+    private static final HereFunctionHandler hereFunctionHandler = new HereFunctionHandler();
+    private boolean containsHereFunction = false;
+    private boolean locationChangedWhileLoading = false;
+
+    public static HereFunctionHandler getHereFunctionHandler() {
+        return hereFunctionHandler;
+    }
+
+    public boolean getContainsHereFunction() {
+        return containsHereFunction;
+    }
+
+    public void setContainsHereFunction(boolean containsHereFunction) {
+        this.containsHereFunction = containsHereFunction;
+    }
+

```

Comment:
Can you move these get/set methods a bit below the main methods of this Activity? For readability sake it is nice to see the most important methods (`onCreate`, `onResume`, etc) near the top of the class.